server/monitor-types/snmp.js

```diff
@@ -0,0 +1,107 @@
+const { MonitorType } = require("./monitor-type");
+const { UP, DOWN, log } = require("../../src/util");
+const snmp = require("net-snmp");
+
+class SNMPMonitorType extends MonitorType {
+    name = "snmp";
+
+    /**
+     * Checks the SNMP value against the condition and control value.
+     * @param {object} monitor The monitor object associated with the check.
+     * @param {object} heartbeat The heartbeat object to update.
+     * @param {object} _server Unused server object.
+     */
+    async check(monitor, heartbeat, _server) {
+
+        log.debug("monitor", `SNMP: Community String: ${monitor.snmpCommunityString}`);
+        log.debug("monitor", `SNMP: OID: ${monitor.snmpOid}`);
+        log.debug("monitor", `SNMP: Version: ${monitor.snmpVersion}`);
+        log.debug("monitor", `SNMP: Condition: ${monitor.snmpCondition}`);
+        log.debug("monitor", `SNMP: Control Value: ${monitor.snmpControlValue}`);
+
+        const options = {
+            port: monitor.port || '161',
+            retries: monitor.maxretries,
+            timeout: 1000,
+            version: getKey(snmp.Version, monitor.snmpVersion) || snmp.Version2c,
+        };
+
+        function getKey(obj, value) {
```

Comment:
please don't declare functions in functions ^^
Also:
why is this needed? (I don't quite understand the `smnp.ObjectType` part)
=> please explain why you are doing as a comment

Reply:
I'm thinking the best place for this function is in utils.ts/utils.js. I will add a comment explaining the reasoning behind this as well.

Reply:
Resolved with c87ac2f0434e4ed3e23b21592c5d080b83ea0f34 and 09fd816aae0f032bf020571522569c6588cc5af8